Framework/Core/include/Framework/GroupedCombinations.h

```diff
@@ -0,0 +1,300 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef FRAMEWORK_PAIR_H
+#define FRAMEWORK_PAIR_H
+
+#include "Framework/ASoAHelpers.h"
+#include "Framework/GroupSlicer.h"
+#include "Framework/StringHelpers.h"
+
+#include <iostream>
+
+namespace o2::framework
+{
+
+// Define the type of a tuple interleaved from 2 tuples
+// Base struct without specialization
+template <typename T1, typename T2>
+struct InterleavedTupleType {
+  using interleaved_pack_t = typename interleave_pack<T1>::template with<T2>::type;
+  template <typename... Is>
+  struct TupleType {
+    using type = std::false_type;
+  };
+  template <std::size_t... Is>
+  struct TupleType<std::index_sequence<Is...>> {
+    using type = std::tuple<pack_element_t<Is, interleaved_pack_t>...>;
+  };
+  using type = typename TupleType<std::make_index_sequence<2>>::type;
+};
+
+// Proper specialization with tuples
+template <typename... T1s, typename... T2s>
+struct InterleavedTupleType<std::tuple<T1s...>, std::tuple<T2s...>> {
+  using interleaved_pack_t = typename interleave_pack<T1s...>::template with<T2s...>::type;
+  template <typename... Is>
+  struct TupleType {
+    using type = std::false_type;
+  };
+  template <std::size_t... Is>
+  struct TupleType<std::index_sequence<Is...>> {
+    using type = std::tuple<pack_element_t<Is, interleaved_pack_t>...>;
+  };
+  using type = typename TupleType<std::make_index_sequence<sizeof...(T1s) + sizeof...(T2s)>>::type;
+};
+
+// Create an instance of a tuple interleaved from given tuples
+template <typename... T1s, typename... T2s, std::size_t... Is>
+auto interleaveTuplesImpl(std::tuple<T1s...>& t1, std::tuple<T2s...>& t2, std::index_sequence<Is...>)
+{
+  return std::tuple_cat(std::make_tuple(std::get<Is>(t1), std::get<Is>(t2))...);
+}
+
+template <typename... T1s, typename... T2s>
+auto interleaveTuples(std::tuple<T1s...>& t1, std::tuple<T2s...>& t2)
+{
+  return interleaveTuplesImpl(t1, t2, std::index_sequence_for<T1s...>());
+}
+
+// Functions to create a tuple from N runs of a function that returns a value
+template <std::size_t I, typename R, typename C, typename... Args>
+R execFunctionWithDummyIndex(R (C::*f)(Args...), C& obj, Args... args)
```

Comment:
There is already functions which do something like this elsewhere. Check Framework/FunctionalHelpers.h.

Reply:
I am not sure this does what we want. Or, at least I don't see how this enables repeated calls of a function?

Reply:
Same here. Please, address the comment about `pack_to_tuple` and the we discuss again.